security/RFP.md

```diff
@@ -0,0 +1,241 @@
+
+The Coordinated Hydra Head V1 specification defines not only the overall life-cycle of the Hydra Head, but also the layer 2 (L2) protocol logic. In the `hydra-node`, the `HeadLogic` layer implements this protocol logic.
+
+Assess that the Hydra node logic layer code faithfully implements the off-chain logic described in the Coordinated Hydra Head V1 specification.
+The outcome of the review should include, bo not being limited to:
+* identify any discrepancy with the off-chain logic from the specification;
+* for any such discrepancy, what security property could be impacted;
+* a review of the tests applied to the Hydra node logic layer and, in particular, any situation that would not be covered by the tests.
+
+This review should consider fair management of the input and output to and from the Hydra node logic layer.
+
+See [Hydra.HeadLogic](https://hydra.family/head-protocol/haddock/hydra-node/Hydra-HeadLogic.html)
+
+### Hydra Head protocol implementation is immune to attacks via chain transactions
+
+The `hydra-node` chain layer is in charge of observing transactions posted on the Cardano blockchain. _Relevant_ observations are then processed in the logic layer, to evolve the internal head status.
+
+Review the Hydra node chain and logic layers in presence of invalid, unexpected or maliciously forged transactions.
+The outcome of the review should include, but not being limited to:
+* identification of transactions or a suite of transactions that would mislead a `hydra-node` into an incorrect head state
+* identification of transactions or a suite of transactions that would lead to a denial of service for the `hydra-node`
+* identification of transactions or a suite of transactions that would invalidate one of the security properties of the Coordinated Hydra Head V1 specification.
+
+### Hydra Head protocol implementation is immune to attacks via network
+
+The `hydra-node` network layer is in charge of sending/receiving messages to/from other Hydra nodes. _Relevant_ messages are then processed in the logic layer, to evolve the internal head status.
+
+Review the Hydra node network and logic layers in presence of invalid, unexpected or maliciously forged network messages.
+The outcome of the review should include, but not being limited to:
+* identification of network attack that would mislead a hydra node to an incorrect head state
+* identification of network attack that would lead to a denial of service for the hydra node
+* identification of network attack that would invalidate one of the security properties of the Coordinated Hydra Head V1 specification.
+
+TODO: should we explicitly state that we are not interested in network denial of service attack? In the assumptions, should we consider only that locally submitted transaction will end up reaching the cardano network or should we enforce stronger network propoerties? For instance, if a node totally loose its network connection, is it OK? For how long is it OK?
```

Comment:
About the network, we should double check what is in the original paper

Reply:
Replaced the TODO with this:
```
Generic denial of service attack against the node itself is out of scope of this audit. One should only focus on specific hydra network layer attacks that could generate a Denial of Service because of an issue in the hydra-node itself.
```

Reply:
Also, in the assumptions, we ruled out attack that would imply fully isolating the node from the network. Because, of course, this would prevent the node from seeing a malicious close transaction and contest it. But... that's an obvious one we don't want an auditor to waste time on.